lib/v8.js

```diff
@@ -15,5 +15,19 @@
 'use strict';
 
 var v8binding = process.binding('v8');
-exports.getHeapStatistics = v8binding.getHeapStatistics;
+
+exports.getHeapStatistics = function() {
+  var data = {};
```

Comment:
This is done incorrectly. The `data` object should be allocated memory once and then reused for every request. There are two ways to do this. The fastest would be to create a `Persistent<Object>` on `Environment` and write the values to that. Another way is to simply get the data off the `Object` by passing it in. I'll comment with more implementation details.

This has a memory leak since it overrides the memory allocated to the object without actually free'ing it.
```
$ /usr/bin/time ./iojs /tmp/get-heap-stats.js 
2377.3 ns/op
2.45user 0.03system 0:02.48elapsed 100%CPU (0avgtext+0avgdata 242016maxresident)k
```
Using this script:
```javascript
var v8 = require('v8');
var ITER = 1e6;

var t = process.hrtime();
for (var i = 0; i < ITER; i++) {
  v8.getHeapStatistics()
}
t = process.hrtime(t);

console.log(((t[0] * 1e9 + t[1]) / ITER).toFixed(1) + ' ns/op');
```